pkg/klusterlet/clusterinfo/defaultInfo_sync.go

```diff
@@ -18,22 +19,32 @@ func (s *defaultInfoSyncer) sync(ctx context.Context, clusterInfo *clusterv1beta
 	if err != nil {
 		return fmt.Errorf("failed to list clusterClaims error:%v ", err)
 	}
+
+	var k8sID string
 	for _, claim := range claims {
 		value := claim.Spec.Value
 		switch claim.Name {
 		case clusterclaim.ClaimOCMConsoleURL:
 			clusterInfo.Status.ConsoleURL = value
 		case clusterclaim.ClaimOCMKubeVersion:
 			clusterInfo.Status.Version = value
+		// use id.openshift.io as cluster ID for ocp clusters
 		case clusterclaim.ClaimOpenshiftID:
 			clusterInfo.Status.ClusterID = value
+		case clusterclaim.ClaimK8sID:
+			k8sID = value
```

Comment:
how about 
```
		case clusterclaim.ClaimK8sID:
			if clusterInfo.Status.KubeVendor == clusterclaim.ProductMicroShift {
		               clusterInfo.Status.ClusterID = value
	                 }
```

Reply:
clusterInfo.Status.KubeVendor may not have value at this time, it gets value at line 37.

Reply:
ok